snake.js

```diff
@@ -2,14 +2,67 @@ const BOARD_X = 40;
 const BOARD_Y = 20;
 const MODULE_SIZE = 20;
 
+class GameBoard {
+    constructor(gameCanvas, color) {
+        this.gameContext = gameCanvas.getContext('2d');
+        this.model = gameCanvas.cloneNode();
+        this.ctx = this.model.getContext('2d');
+        this.ctx.lineWidth = 1;
+        this.ctx.strokeStyle = color;
+        this.ctx.fillStyle = 'white';
+        this.build();
+    }
+    build() {
+        this.ctx.fillRect(0, 0, this.model.width, this.model.height);
+
+        for (let x = MODULE_SIZE + 1.5; x < this.model.width; x += MODULE_SIZE + 1) {
+            this.ctx.moveTo(x, 0);
+            this.ctx.lineTo(x, this.model.height);
+        }
+
+        for (let y = MODULE_SIZE + 1.5; y < this.model.height; y += MODULE_SIZE + 1) {
+            this.ctx.moveTo(0, y);
+            this.ctx.lineTo(this.model.width, y);
+        }
+
+        this.ctx.stroke();
+    }
+    draw() {
+        this.gameContext.drawImage(this.model, 0, 0);
+    }
+}
+
+class SnakeBody
+{
+    constructor()
+    {
+        this.model=document.createElement("canvas");
+        this.canvas=canvas;
+        this.model.width = MODULE_SIZE;
+        this.model.height = MODULE_SIZE;
+        this.ctx = this.model.getContext('2d');
+        this.bodyParts = 1;
+        this.build();
+    }
+    build()
+    {
+        this.ctx.fillStyle = "rgb(0,12,0)";
+        this.ctx.fillRect(0,0,MODULE_SIZE, MODULE_SIZE);
```

Comment:
We have to agree on some `model` style. We don't want to have our snake modules to look like filled rectangles after all.